rusk/src/lib/chain/rusk.rs

```diff
@@ -480,6 +484,13 @@ fn accept(
     ))
 }
 
+fn strip_off_bytecode(tx: &PhoenixTransaction) -> Option<PhoenixTransaction> {
+    let _ = tx.payload().contract_deploy()?;
+    let mut tx_clone = tx.clone();
+    tx_clone.strip_off_bytecode();
+    Some(tx_clone)
+}
```

Comment:
I'd be really happy to eliminate cloning here, yet method `execute` receives an immutable copy of tx, and to modify it I need to clone. On the other hand, to pass a mutable tx into `execute` is impossible, as `execute` is called from a tx processing loop and txs are needed upstream in an "untouched" form. Hence, cloning is inevitable, but we could think of a smarter way of cloning so that the bytecode itself is not cloned. WTS.

Reply:
OK - optimized cloning so that the bytecode part is not being cloned. Other fields need to be cloned though, as we need a clone of the entire transaction for reasons mentioned above.